mlir/lib/Dialect/MIOpen/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -335,7 +176,7 @@ struct BlockwiseGemmRewritePattern : public OpRewritePattern<BlockwiseGemmOp> {
     auto loopOp = b.create<AffineForOp>(loc, 0, loopIteration);
 
     // inside the main loop.
-    auto lb = OpBuilder::atBlockTerminator(loopOp.getBody());
+    auto lb = OpBuilder::atBlockTerminator(loopOp.getBody(), b.getListener());
```

Comment:
Do we really depend on methods in `OpBuilder::Listener`, such as `notifyOperationInserted` / `notifyOperationRemoved`?

Reply:
The pattern rewriting infrastructure does, I'm pretty sure

Reply:
Compared with the original implementation, why didn't we need it before, but need it now?

Reply:
Because not doing that was wrong

Reply:
Could you shed some lights on the reason? Quite a few call sites in `lib/Dialect/Affine/Utils/LoopUtils.cpp` do so without the `listener` argument. From the interface the argument is optional, and in the PR I don't see any overrides to `notifyOperationInserted/notifyOperationRemoved`.

Reply:
The code in `LoopUtils` isn't meant to run within a pattern rewrite